### ISSUE TITLE
Fatal: Unable to initialize console - unable to retrieve modules

### ISSUE THREAD

**Issue by RichAyotte** [ethereum/go-ethereum#2155]
When trying to connect to a remote server, the following error occurs.

```bash
$ geth attach rpc:http://192.168.1.12:8545
Fatal: Unable to initialize console - unable to retrieve modules
```
Local version
```bash
$ geth version
Geth
Version: 1.4.0-unstable (1.4.0+64SNAPSHOT20160126130620wily-0ubuntu1)
Protocol Versions: [63 62 61]
Network Id: 1
Go Version: go1.5.1
OS: linux
GOPATH=
GOROOT=/usr/lib/go
```

Remove version
```bash
$ geth version
Geth
Version: 1.3.3
Protocol Versions: [63 62 61]
Network Id: 1
Go Version: go1.5.1
OS: linux
GOPATH=
GOROOT=/usr/lib/go
```

**Comment by karalabe** [comment]
Latest develop includes a full rewrite of the RPC layer, so there may be some quirks still lurking in there. I'll take a look in the morning.

**Comment by karalabe** [comment]
Hey, I've checked it out. 1.4 introduces a new RPC internals, with support for push notifications among others. To make things cleaner, we've changed the name of some internal RPC method calls. Because of this, 1.4 queries available modules differently than 1.3.x did, hence why the two can't talk to each other.

However, as we don't intend currently to support cross version attaches, these kinds of errors aren't really a priority. The RPC changes too much currently to make guarantees beyond the officially speced API calls.

To work around this, simply use the same version of geth on both the "server" and the "client" side :)
